### PR TITLE
[Custom Trainer] Add a verbose help output structure to the CLI

### PR DESCRIPTION
**Pull request by harimkang**
## Description

Addition of the HelpFormatter class to manage argument exposure based on rich help formats and verbosity level.

## Changes

As jsonargparse was applied to Anomalib 1.0, 
Verbosity-CLI was applied to reduce issues with exposing too much information, similar to the OTX 2.0 CLI.

1. Enable rich-based help formatter.
> anomalib -h
image

2.  Enable Verbosity-Level (fit, validate, test, predict)
- Verbosity 0 : Quick-Guide only
> anomalib fit -h
image

- Verbosity 1: Quick-Guide + Required Arguments
> anomalib fit -h -v
image
image

- Verbosity 2: All Arguments (Same with existing)
> anomalib fit -h -vv

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

image

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

**Comment by ashwinvaidya17** [comment]
Wow that was quick!

File: src/anomalib/utils/cli/help_formatter.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    "fit": Engine.fit,
+    "validate": Engine.validate,
+    "test": Engine.test,
+    "predict": Engine.predict,
+}
+
+
+def pre_parse_arguments() -> dict:
+    """Pre-parse arguments.
+
```
Comment (ashwinvaidya17) [inline src/anomalib/utils/cli/help_formatter.py:34]: Can you also add an explination on what this method does?

**Review by ashwinvaidya17 — APPROVED** [review]
A few minor comments but other than that it looks really cool.

**Comment by djdameln** [comment]
@ashwinvaidya17 we always seem to get the `PrecisionRecallCurve` warning when calling `anomalib -h`, (see Harim's screenshots). Any idea why this is the case? Ideally we should not be running metrics-related code when running with just the `--help` argument right? Maybe this also contributes to the slowness of `anomalib --help`?

**Comment by harimkang** [comment]
> Maybe this also contributes to the slowness of `anomalib --help`?

@djdameln @ashwinvaidya17 @samet-akcay 
I looked at two things related to your comment.
1. torchmetrics warning log -> it comes from `from anomalib.utils.callbacks.thresholding import _ThresholdCallback`
But I don't have an idea for getting rid of this log yet.

2. Slowness of the CLI
I simply ran a scan on anomalib.utils.cli with the import profiling tool.

import anomalib.utils.cli : 3.82s (Total Time)
-> import lightning : 3.08s
              - import lightning.app: 2.02s
-> import anomalib.data: 0.37s
-> import anomalib.engine: 0.31s
-> import anomalib.utils.benchmarking: 0.04s

What I'm thinking from the above results is that cli is spending the most time on the lightning import, so it may not be a problem inside anomalib. But still, it would be nice if we could find a way to reduce the lightning import time in CLI.

**Review by samet-akcay — APPROVED** [review]

**Review by djdameln — APPROVED** [review]

**Pull request merged**